tv_apps/smart-home/js/home.js

```diff
@@ -793,28 +793,34 @@
     },
 
     updateClock: function() {
+      var formatter = new Intl.DateTimeFormat(navigator.languages, {
+        hour: 'numeric',
+        minute: 'numeric',
+        hour12: window.navigator.mozHour12
+      });
       var now = new Date();
-      var _ = navigator.mozL10n.get;
-      var use12Hour = window.navigator.mozHour12;
 
-      var f = new navigator.mozL10n.DateTimeFormat();
+      var parts = formatter.formatToParts(now);
 
-      var timeFormat = use12Hour ? _('shortTimeFormat12') :
-                                   _('shortTimeFormat24');
-      // remove AM/PM and we use our owned style to show it.
-      timeFormat = timeFormat.replace('%p', '').trim();
-      var formatted = f.localeFormat(now, timeFormat);
+      var dayperiod = '';
 
-      this.timeElem.innerHTML = formatted;
-      this.timeElem.dataset.ampm = use12Hour ? f.localeFormat(now, '%p') : '';
+      var timeWithoutDayPeriod = parts.map(({type, value}) => {
+        if (type === 'dayperiod') {
+          dayperiod = value;
+          return '';
+        }
+        return value;
+      }).join('');
+
+      if (this.timeElem) {
```

Comment:
I'm using this here because unit test loads the file before creating the DOM elements so the `this.timeElem` is not available in the test unit. Before it was not tested because it was wrapped in `mozL10n.ready` which was not fired, but now it has to be guarded this way until the test is fixed